docs/specification.rst

```diff
@@ -1,11 +0,0 @@
-Raiden Specification
```

Comment:
We still need to have this done at some point, but yes apart from the TODO it's empty so we can recreate the file at any point if you feel strongly for deleting the file.